try: ci: bump cmake minimum version

gastmaier opened the pull request:
## Pull Request Description

Compatibility with CMake < 3.5 has been removed from CMake. Update CMake files dependencies to >= 3.5.
For doxygen, it is release 3.12 the new minimum version, due to the support to C++20 modules.
https://github.com/doxygen/doxygen/commit/6d80fc7e5d03c259b1a7280972e0b28884217655

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [ ] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [ ] I have complied with the [Submission Checklist](http://analogdevicesinc.github.io/no-OS/contributing.html#submission-checklist)
- [ ] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [ ] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

gastmaier committed:
ci: bump cmake minimum version to 3.5

Compatibility with CMake < 3.5 has been removed from CMake.
Update CMake files dependencies to >= 3.5.

Signed-off-by: Jorge Marques <jorge.marques@analog.com>

gastmaier commented:
Now fails around
```
patching file ./src/docparser.cpp
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
2 out of 2 hunks ignored -- saving rejects to file ./src/docparser.cpp.rej

```
https://dev.azure.com/AnalogDevices/no-OS/_build/results?buildId=47531&view=logs&j=f6608c7f-089a-573f-f706-783559be9c63&t=37880869-df78-5570-2d40-f1d59ee40cae

buha commented:
This line may need to be removed:
```
ci/run_build.sh
95:    patch ./src/docparser.cpp < ${TOP_DIR}/ci/patch_doxy.patch
```

buha committed:
ci: bump doxygen version to 3.12

For doxygen, it is release 3.12 the new minimum version,
due to the support to C++20 modules.

Remove the patching that we were previously doing on the
1.8.17 version.

Signed-off-by: Darius Berghe <darius.berghe@analog.com>

buha commented:
@gastmaier i took the doxygen change you made and made a new commit from it which also removes the patching, i'm waiting to see if it fixes that new error

buha commented:
unfortunately (of fortunately? :D) this new version gets past those issues but finds lots of issues like this, which the previous version was ignoring for some reason:
```
Generating caller graph /home/vsts/work/1/s/drivers/accel/adxl345/adxl345.c:250: warning: found documented return type for adxl345_get_g_xyz that does not return anything
```

These are valid points and should be fixed.

amiclaus reviewed (commented):
@gastmaier I added https://github.com/analogdevicesinc/no-OS/pull/2528 which updates the doxygen version to the newest one including the patch handling/fixing new detected errors.

buha commented:
i merged @amiclaus 's pull request #2528 and it fixes the problem so I'm closing this one.
